### ISSUE TITLE
[#85 M6a] intake_node: BERT fast-path call + intake_handoff emission

### ISSUE THREAD

**Issue by Exios66** [LLM-Mailroom-Services/mailroom-issues#98]
## Parent
Child of [#85](https://github.com/LLM-Mailroom-Services/mailroom-issues/issues/85).

**Related:** [#88](https://github.com/LLM-Mailroom-Services/mailroom-issues/issues/88) (M3 inference path), [#91](https://github.com/LLM-Mailroom-Services/mailroom-issues/issues/91) (M6 continuation), [#90](https://github.com/LLM-Mailroom-Services/mailroom-issues/issues/90) (M5 gate).

## Goal
Wire the ModernBERT fast-path classifier into the pipeline's `intake_node` so the node emits a schema-v1 `intake_handoff` (+ `bert_triage`) onto `DocumentState`, with **zero behavior change until the flag is on**. This is the ingest seam of the intake overhaul; it does **not** yet skip the sorter (that is M6b).

## Host repo
`LLM-Mailroom-Services/Digital-Mailroom` → `packages/llm-mailroom`

## Exact seams
- `src/graph/build_graph.py::intake_node` (line ~540) — after `apply_intake` (clerk, ~586) and the gated LLM intake prep (~589–623), before the manifest write (~644).
- Imports **must be lazy/guarded** (try/except `ImportError` → BERT unavailable → LLM path), because `mailroom-ml` may not be installed in every runtime:
  ```python
  from mailroom_ml.inference import load_bundle, classify_document
  from mailroom_ml.routing import should_bert_intake, build_handoff
  ```

## Implementation sketch
1. Resolve the bundle once per process (module-level cache): `load_bundle()` honors `ML_MODEL_DIR`; wrap → `bundle=None` on any failure (fail-open).
2. Compute `title` from intake metadata (**subject → exhibit_description → EMPTY — never the filename**; the filename label-leak was fixed in the training data 2026-09-20 and must not be reintroduced at inference).
3. `ok, reason = should_bert_intake(doc_text, intake_stats, model_available=bundle is not None)`
   - False reasons: `flag_off` | `empty` | `messy` | `oversize_chars` | `oversize_tokens` | `no_model` → record `routing_reason`, skip BERT.
4. If `ok`: `triage = classify_document(bundle, title, doc_text)` — never raises (fail-open by construction); returns `{status, doc_type, subclass, route, reason, score, quality, guard_failures, artifact_sha, n_windows, calibrated_confidence, subclass_confidence, agreement, margin, ...}`.
5. Build the handoff:
   ```python
   handoff = build_handoff(
       method="bert", routing_path="bert_intake", routing_reason=reason,
       doc_chars=len(doc_text), doc_token_estimate=intake_stats.get("token_estimate", 0),
       doc_type=triage.get("doc_type"), doc_subclass=triage.get("subclass"),
       confidence=triage.get("calibrated_confidence", 0.0),
       quality=triage.get("quality"), guard_failures=triage.get("guard_failures"),
       model_id=MODEL_ID, model_revision=bundle.artifact_sha,
       intake_windows=triage.get("n_windows", 0))
   ```
6. Emit the `intake-ml-triage` span (plan §9) with provenance: `model_id`, `artifact_sha`, `dataset_revision`, `calibration_version`, `label_schema_version`, `window_agreement`.
7. Put on state: `intake_handoff`, `bert_triage`, `bert_routing_reason`.
8. **Always** build a handoff — on the flag-off / BERT-unavailable path use the defaults (`method="deterministic"`, `routing_path="clerk_only"`, `recommended_action="force_llm_sorter"`) so downstream consumers can rely on its presence.

## Acceptance criteria / DoD
- [ ] Flag off (`MAILROOM_BERT_INTAKE=0`) → byte-identical behavior; handoff still emitted with `method=deterministic`, `routing_path=clerk_only`.
- [ ] BERT artifact absent / import error / exception → `method=deterministic`, no crash, span records the failure (fail-open, plan D10).
- [ ] `intake-ml-triage` span carries `model_id` + `artifact_sha` + dataset revision + `window_agreement`.
- [ ] No filename ever reaches the classifier as title (leak guard test).
- [ ] Unit test: stub bundle → deterministic handoff shape asserted.
- [ ] Intake never alone marks a run FAILED.

## Depends on
- [#88](https://github.com/LLM-Mailroom-Services/mailroom-issues/issues/88), [#86](https://github.com/LLM-Mailroom-Services/mailroom-issues/issues/86), [#87](https://github.com/LLM-Mailroom-Services/mailroom-issues/issues/87)

## Blocks
- M6b (after_intake edge)

**Comment by Exios66** [comment]
**Implementation review (2026-09-20) — seams verified against `packages/llm-mailroom` @ Digital-Mailroom b1321f65 sync**

All line anchors from the sketch check out exactly. Deltas and confirmations:

1. **Contract exists upstream.** `mailroom_ml.routing` ships `should_bert_intake` (reasons: `flag_off|empty|messy|oversize_chars|oversize_tokens|no_model`), `build_handoff` (schema v1, exact epic shape), `evaluate_intake_gate` (P1–P7/F1–F7, never raises); `mailroom_ml.inference` ships `load_bundle` (resolution `ML_MODEL_DIR` > override > `artifacts/{pytorch,onnx}/model`, fail-open), `classify_document` (never raises; `{status, doc_type, subclass, route, reason, score, quality, guard_failures, artifact_sha, n_windows, calibrated_confidence, subclass_confidence, agreement, margin}`). Lazy/guarded import is the only path — `mailroom-ml` is not a declared dep (zero references in repo; pyproject has no ml pin; mailroom-ml has no git tags yet — pin a commit SHA when we do declare it).
2. **Insertion point confirmed**: between `apply_intake` (build_graph.py:586) and the LLM gate (591). The gate line `should_llm_intake(doc_text, intake_stats)` (agents/intake.py:183, no `ml_triage` param) must be composed at the call site to honor the BERT route — ⚠️ see child #102: the mailroom-ml `should_llm_intake` currently ignores `BERT_INTAKE_MODE`; default shadow + flag-on would silently skip the sorter. Fix lands in mailroom-ml; this node must also gate sorter-skip on mode anyway.
3. **Manifest merge bug to fix**: `intake_meta` merge (build_graph.py:634–643) is gated on `intake_prep` truthy — a BERT-only run has `intake_prep=None`, so the handoff would be dropped before the manifest write (644). Merge `handoff` into `intake_meta` unconditionally.
4. **New state fields** (none exist): `intake_handoff`, `bert_triage`, `bert_routing_reason`, `classification_method` → add to `DocumentState` (TypedDict, total=False — additive-safe).
5. **`token_estimate`**: `apply_intake` stats carry no `token_estimate` today. `should_bert_intake` needs `stats["token_estimate"]` — add it to intake_stats but OUTSIDE the dojo-parity dict (`src/tests/test_intake.py:83` asserts full-dict equality `our_stats == their_stats` — additive-safe membership is fine, full-dict equality is not).
6. **Title**: `subject → exhibit_description → EMPTY`, never filename (leak guard test required; training data republish 5b72a34 already leak-free).
7. **Span**: emit `intake-ml-triage` as a **span** (not generation — no LLM call; local-inference precedent `normalize-intake`) via `observation()` (tracing.py:117), metadata: model_id, artifact_sha, dataset_revision, calibration_version, label_schema_version, window_agreement; output: available, reason, route, triage_class, confidence, latency_ms; always emit (fail-open shows `available:false` + reason → fail-soft-rate dashboards). Never doc_text (PII discipline).
8. **Hermetic tests**: conftest `_set_test_env` must force `MAILROOM_BERT_INTAKE=0`, pop `ML_MODEL_DIR` + `BERT_INTAKE_MODE` (models = network hazard, cf. `MAILROOM_RELATIONS_EMBEDDINGS=0`).
9. **Bundle pin**: `Lucius-Morningstar/mailroom-modernbert-classifier` @ main `583f1d83e3…` is PyTorch-only (safetensors+heads.pt); no ONNX on the Hub yet — #101 owns export/publish. Start the startup parity gate as **subset-aware** (corporate_record 10 vs live 11 → `certificate_of_formation→other` projection; see #107).

DoD additions: flag-off byte-identical E2E + handoff `method=deterministic/clerk_only` asserted (home: test_intake_agent.py:448 + test_pipeline_e2e.py); stub-bundle handoff shape test; fail-open import-error test; title-leak assert.

**Comment by Exios66** [comment]
**M6a implemented + merged to the standalone mirror (commit `ba7e010`)** — the #98 seam is cut, fail-open, Tier-1-ready.

Landed:
- **`src/agents/bert_intake.py`** (new): `run_bert_intake()` procedural lane — lazy-guarded `mailroom_ml` import, **handoff contract v1** (`{available, reason: flag_off|no_package|no_model|error|ok, method: bert|deterministic, routing_path: fast_path|clerk_only}` + flattened runner fields), `_write_bert_debug()` capture under `data/debug/bert_intake/` (kill-switch `MAILROOM_BERT_DEBUG=0`), `format_bert_type_prior()` for the Tier-1 prior channel.
- **`src/graph/build_graph.py`**: intake_node runs the lane inside a new **`intake-ml-triage` SPAN** (span type — no LLM call; curated provenance metadata; emitted even when flag-off); **the manifest-merge bug is fixed** — the bert handoff merges UNCONDITIONALLY (the old `intake_prep` gate dropped it on BERT-only runs); state carries `bert_triage` + `intake_handoff`; classify_node composes `format_bert_type_prior` onto the **existing `intake_prior=` channel** when route=fast_path (vendored sorter_v14 never mutated, zero new LLM calls) — the #108 Tier-1 landing seam is live.
- **`state.py`**: 2 optional fields. **conftest**: `MAILROOM_BERT_INTAKE=0` + pops. **`.env.example`/AGENTS.md**: opt-in knobs + lane doc.
- **Tests**: 16 in `test_bert_intake.py` — fail-open matrix (flag-off never imports the package, no_package/no_model/error/non-dict/TypeError-fallback), happy-path handoff shape, prior formatting + exact call-site composition, node-level integration **pinning the old manifest-merge bug**, error-never-breaks-run.

Gates: `pytest src/tests` → **1068 passed, 37 skipped** (incl. notebook goldens 79/79; the only pre-change failure was a missing `[notebooks]` extra in my fresh venv — environmental, not code).

⚠️ **SYNC OBLIGATION (DMR-074)**: the commit lives in the standalone mirror only (no monorepo checkout on this machine). Before the next propagation: monorepo-side `sync_packages.py pull --squash`, then `push --package llm-mailroom --patch --verify-suite`. Do NOT push the standalone mirror directly past this.

Blocked-on update: this seam unblocks #99 (after_intake router — Row 2a/2b split) and #108 (Tier-1 prior consumption is already plumbed; the router split + scoped max_tokens land next).

**Comment by Exios66** [comment]
## ✅ Closed — implemented, synced, propagated

Evidence chain, complete:
1. **Implementation**: commit `ba7e010` (#85 M6a) — `agents/bert_intake.py` + handoff-merge fix + Tier-1 prior seam + 16 tests. Full suite **1068 passed, 37 skipped** (incl. notebook goldens).
2. **Propagation (DMR-074)**: imported into the Digital-Mailroom monorepo at `7d6bde2c` (merge-base-resolved vs the monorepo-side uv-workspace fix `243d10f0`; pyproject adaptation preserved; all 7 content files byte-identical to the tested tree). Cursor re-baselined; `sync_packages.py status` now shows llm-mailroom **SYNCED**.
3. **Downstream units riding this seam**: #108 (Tier-1 prior consumption — live), #99 (after_intake router — next), #90 (sorter-skip gate — after #102/#103 contract fixes), #109 (Docker deployment matrix — landed `1bdb1dbb`, propagated `417d5a08`).

The old `intake_prep`-gated manifest merge bug this card surfaced is pinned by the node-level integration test (`test_node_merges_handoff_into_manifest`).

**Comment by Exios66** [comment]
Post-close runtime note (2026-09-21): the lane's production call `classify_document(doc_text, filename=...)` never matched the real mailroom-ml signature (`(bundle, title, doc_text, ...)`) — every real ingestion fell through to the clerk ERROR handoff; the handoff DoD was met only under the TypeError dance. Fixed with a first-class seam: `classify_document_default(doc_text, *, filename, ...)` (mailroom-ml `628d80a` — default-bundle resolution + process cache + fail-open marker vocabulary) and the lane prefers it with the legacy fallback (llm-mailroom `8384347f`, +1 compat test). The tier lanes (#90/#91/#108) now sit on a handoff that can actually exist in production.
